src/kv/test/kv_snapshot.cpp

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+
+#include "kv/kv_serialiser.h"
+#include "kv/store.h"
+#include "kv/test/null_encryptor.h"
+#include "kv/tx.h"
+
+#include <doctest/doctest.h>
+
+struct MapTypes
+{
+  using StringString = kv::Map<std::string, std::string>;
+  using NumNum = kv::Map<size_t, size_t>;
+};
+
+TEST_CASE("Simple snapshot" * doctest::test_suite("snapshot"))
+{
+  kv::Store store;
+  auto& string_map = store.create<MapTypes::StringString>(
+    "string_map", kv::SecurityDomain::PUBLIC);
+  auto& num_map =
+    store.create<MapTypes::NumNum>("num_map", kv::SecurityDomain::PUBLIC);
+
+  INFO("Apply transactions to original store");
+  {
+    kv::Tx tx1, tx2;
+    auto view_1 = tx1.get_view(string_map);
+    auto view_2 = tx2.get_view(num_map);
+
+    view_1->put("foo", "bar");
+    view_2->put(42, 123);
+
+    REQUIRE(tx1.commit() == kv::CommitSuccess::OK); // Committed at 1
+    REQUIRE(tx2.commit() == kv::CommitSuccess::OK); // Committed at 2
```

Comment:
This looks unnecessarily like conflicting transactions. because we interleave the operations. Since these are really applied in series, and we're not trying to test interleaved operations, can we write these in sequence? ie

```
kv::Tx tx1;
auto view_1 = ...;
view_1->put(...);
REQUIRE(tx1...);

kv::Tx tx2;
auto view2 = ...;
...
```